packages/flutter_tools/test/general.shard/asset_bundle_test.dart

```diff
@@ -180,6 +180,73 @@ flutter:
       FileSystem: () => testFileSystem,
       ProcessManager: () => FakeProcessManager.any(),
     });
+
+    group('on Unix', () {
+      testUsingContext('ignores hidden files', () async {
+        globals.fs
+          .file(globals.fs.path.join('assets', 'foo', 'bar.txt'))
+          .createSync(recursive: true);
+        globals.fs
+          .file(globals.fs.path.join('assets', 'foo', '.hidden_unix_file'))
+          .createSync();
+        // Create a directory in the same path to test that we're only looking at File
+        // objects.
+        globals.fs
+            .directory(globals.fs.path.join('assets', 'foo', 'bar'))
+            .createSync();
+        globals.fs.file('pubspec.yaml')
+          ..createSync()
+          ..writeAsStringSync(r'''
+            name: example
+            flutter:
+              assets:
+                - assets/foo/
+            '''
+          );
+          globals.fs.file('.packages').createSync();
+          final AssetBundle bundle = AssetBundleFactory.instance.createBundle();
+          await bundle.build(manifestPath: 'pubspec.yaml');
+          // Expected assets:
+          //  - asset manifest
+          //  - font manifest
+          //  - license file
+          //  - assets/foo/bar.txt
+          expect(bundle.entries.length, 4);
+          expect(bundle.needsBuild(manifestPath: 'pubspec.yaml'), false);
+      }, overrides: <Type, Generator>{
+        FileSystem: () => testFileSystem,
+        ProcessManager: () => FakeProcessManager.any(),
+      });
+
+      testUsingContext('trace text contains info regarding not included hidden file', () async {
+        globals.fs
+          .file(globals.fs.path.join('assets', 'foo', 'bar.txt'))
+          .createSync(recursive: true);
+        globals.fs
+          .file(globals.fs.path.join('assets', 'foo', '.hidden_unix_file'))
+          .createSync();
+        // Create a directory in the same path to test that we're only looking at File
+        // objects.
+        globals.fs
+          .directory(globals.fs.path.join('assets', 'foo', 'bar'))
+          .createSync();
+        globals.fs.file('pubspec.yaml')
+          ..createSync()
+          ..writeAsStringSync(r'''
+name: example
+flutter:
+  assets:
+    - assets/foo/
+''');
+        globals.fs.file('.packages').createSync();
```

Comment:
To be valid this needs to have a `\n`. But I'm not sure if it is needed?

Reply:
Not sure what you mean exactly. I just followed other tests when creating this one.